### PR TITLE
Audio cog playlist_extend implementation

### PR DESCRIPTION
**Pull request by frrleone**
Implemented playlist_extend in audio cog

Changed _parse_yt_playlist and _parse_sc_playlist to return error messages when the playlist they are passed is invalid

### CODE REVIEW

File: cogs/audio.py [hunk 1]
```diff
@@ -763,13 +763,16 @@ def _match_any_url(self, url):
...
-                playlist.append(entry.url)
-
+        try:
+            for entry in d.song.entries:
+                if entry["url"][4] != "s":
+                    song_url = "https{}".format(entry["url"][4:])
+                    playlist.append(song_url)
+                else:
+                    playlist.append(entry.url)
+        except:
```
Comment (tekulvw) [inline cogs/audio.py:773]: Don't have catch all exception handlers here. Also try to minimize the amount of code inside of a `try/except` and explicitly catch exceptions.

File: cogs/audio.py [hunk 2]
```diff
@@ -780,15 +783,19 @@ def _match_any_url(self, url):
...
+                try:
+                    song_url = "https://www.youtube.com/watch?v={}".format(
+                        entry['id'])
+                    playlist.append(song_url)
+                except AttributeError:
+                    pass
+                except TypeError:
+                    pass
+
+        except:
```
Comment (tekulvw) [inline cogs/audio.py:797]: Same thing down here.

**Review by tekulvw — CHANGES_REQUESTED** [review]

**Comment by Kowlin** [comment]
Changes & merge issue has to be resolved before merging.

**Comment by Tobotimus** [comment]
Closing due to staleness and lack of response.

Can always be re-opened if the changes are made and it is ready to be merged again.

**Pull request closed without merging**